app/src/main/java/org/simple/clinic/login/pin/LoginPinScreen.kt

```diff
@@ -72,10 +74,19 @@ class LoginPinScreen(context: Context, attrs: AttributeSet) : RelativeLayout(con
       RxTextView.editorActions(pinEditText) { it == EditorInfo.IME_ACTION_DONE }
           .map { PinSubmitClicked() }
 
-  private fun backClicks() =
+  private fun backClicksFromUi() =
       RxView.clicks(backButton)
           .map { PinBackClicked() }
 
+  private fun backClicksFromSystemBack() =
+      Observable.create<PinBackClicked> { emitter ->
+        screenRouter.registerBackPressInterceptor(object : BackPressInterceptor {
+          override fun onInterceptBackPress(callback: BackPressInterceptCallback) {
+            emitter.onNext(PinBackClicked())
+          }
+        })
```

Comment:
You forgot to unregister the interceptor on dispose :)

Reply:
Whoops. Will fix.